timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/AsyncLockService.java

```diff
@@ -165,6 +174,11 @@ private AsyncResult<HeldLocks> acquireImmutableTimestampLock(UUID requestId, lon
         return lockAcquirer.acquireLocks(requestId, OrderedLocks.fromSingleLock(immutableTsLock), TimeLimit.zero());
     }
 
+    private AsyncResult<HeldLocks> acquireNamedMinTimestampLock(String timestampName, UUID requestId, long timestamp) {
```

Comment:
Could we in a FLUP experiment with having MinTimestampName be a union of ReservedTimestampName (which is the immutable timestamp) and just a string wrapper?

We'd really ideally get rid of this weird duplication of APIs, because it's exactly the same.

Java 17 sealed type would be nice for this